src/v/cluster/health_monitor_types.cc

```diff
@@ -148,14 +148,18 @@ std::ostream& operator<<(std::ostream& o, const partitions_filter& filter) {
 namespace reflection {
 
 template<typename T>
-void read_and_assert_version(std::string_view type, iobuf_parser& parser) {
+int8_t read_and_assert_version(
+  std::string_view type,
+  iobuf_parser& parser,
+  int8_t expect_version = T::current_version) {
     auto version = adl<int8_t>{}.from(parser);
     vassert(
-      version <= T::current_version,
+      version <= expect_version,
       "unsupported version of {}, max_supported version: {}, read version: {}",
       type,
       version,
       T::current_version);
```

Comment:
I think it's worth swapping these two fields to log the correct message

Reply:
I'm doing that cleanup on dev here https://github.com/redpanda-data/redpanda/pull/4195/files

Avoiding pushing a revision to this PR so that the CI can run through and be ready to merge.